src/wayland-eglsurface.c

```diff
@@ -191,8 +195,19 @@ wlEglSendDamageEvent(WlEglSurface *surface, struct wl_event_queue *queue)
                           surface->dy);
     }
 
-    wl_surface_damage(surface->wlSurface, 0, 0,
-                      surface->width, surface->height);
+    if (n_rects == 0) {
+        wl_surface_damage(surface->wlSurface, 0, 0,
+                          surface->width, surface->height);
```

Comment:
This is still incorrect in the presence of surface transforms (HiDPI scale, rotation, etc). In this case just use `UINT_MAX` as the w/h and it'll get clipped.

Reply:
Fixed, and thanks for the review!